ravenframework/TSA/STL.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Created on July 7, 2023
+@author: j-bryan
+
+Detrending with STL decomposition
+"""
+
+import numpy as np
+
+from .TimeSeriesAnalyzer import TimeSeriesTransformer, TimeSeriesGenerator
+from ..utils import importerUtils, InputTypes, InputData, xmlUtils
+statsmodels = importerUtils.importModuleLazy('statsmodels', globals())
```

Comment:
I wasn't sure why exactly it was done this way, but statsmodels is imported like this in the `ARMA` class as well. I followed the pattern because I saw something about using lazy importing of `statsmodels` (and a few other libraries) as standard for RAVEN. Do you know what's going on there that's not accomplished by an import in the `fit` method?